File: apps/src/chat/backend/packets/websocket/global_join_packet.py
from ..packet import Packet

class GlobalJoinPacket(Packet):

    def __init__(self, username: str):
        self._username = username

    def to_data(self):
        return {
            "username": self._username,
            "type": "global_join"
        }

    @staticmethod
    def from_data(data):
        return GlobalJoinPacket(
            data["username"]
        )

File: apps/src/chat/frontend/js/packets/websocket/GlobalJoinPacket.js
export default class GlobalJoinPacket {

    constructor(username = ""){
        this.username = username;
    }

    toData(){
        return {
            username: this.username,
            type: "global_join"
        }
    }

    static fromData(data){
        return new GlobalJoinPacket(data["username"]);
    }

    getUsername(){
        return this.username;
    }
}